### PR TITLE
Bugfix and enhancement from customers

### PR DESCRIPTION
**Pull request by taohe1012**
# Description
Bugfix and enhancement from customers:
https://github.com/dell/dell-terraform-providers/issues/9
https://github.com/dell/dell-terraform-providers/issues/10
https://github.com/dell/dell-terraform-providers/issues/11
https://github.com/dell/dell-terraform-providers/issues/12
https://github.com/dell/dell-terraform-providers/issues/13

Solutions:
1. Update Read action to make cluster-side updates invisible to terraform for User, User Groups, Groupnet and Access Zone resources.
2. Remove hard codes of auth providers for Access Zone resource.
3. Add validation for different quota type for Quota resource.
4. Change owner and groups fields to Optional for File System resource.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/dell-terraform-providers/issues/9 |
|https://github.com/dell/dell-terraform-providers/issues/10 |
|https://github.com/dell/dell-terraform-providers/issues/11 |
|https://github.com/dell/dell-terraform-providers/issues/12 |
|https://github.com/dell/dell-terraform-providers/issues/13 |

# ISSUE TYPE

- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
User Resource
User Group resource
Groupnet resource
Access Zone resource
Quota resource
File System resource

##### OUTPUT
All FT and AT are passed for above resource.
And their test coverage are above 80%.
image
image
image
image
image
image

image
image
image
image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

File: powerscale/helper/groupnet_helper.go [hunk 1]
```diff
@@ -56,21 +56,30 @@ func UpdateGroupnetDataSourceState(ctx context.Context, groupnetState *models.Gr
...
+	originModel := *groupnetModel
 
-	groupnetModel.ID = types.StringValue(*groupnetResponse.Id)
-	groupnetModel.Name = types.StringValue(*groupnetResponse.Name)
-	groupnetModel.AllowWildcardSubdomains = types.BoolValue(*groupnetResponse.AllowWildcardSubdomains)
-	groupnetModel.DNSCacheEnabled = types.BoolValue(*groupnetResponse.DnsCacheEnabled)
-	groupnetModel.ServerSideDNSSearch = types.BoolValue(*groupnetResponse.ServerSideDnsSearch)
+	if err = CopyFields(ctx, groupnetResponse, groupnetModel); err != nil {
+		return
+	}
```
Comment (NavisJ) [inline powerscale/helper/groupnet_helper.go:63]: If CopyFields returns no error, the following code will not execute. Is that expected?
Cause I think it in most cases will not return error unless passing in parameters of wrong data type.

Comment (taohe1012) [reply powerscale/helper/groupnet_helper.go:63]: @NavisJ thinking no matter errors, CopyFields should always be executed and the groupnetResponse is already parsed to groupnetModel.

Comment (NavisJ) [reply powerscale/helper/groupnet_helper.go:63]: yes. You are right. I made a mistake with the function.

**Review by NavisJ — APPROVED** [review]

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Comment by taohe1012** [comment]
run e2e test

**Review by forrestxia — APPROVED** [review]

**Pull request merged**